virtual_rainforest/models/plants/canopy.py

```diff
@@ -133,49 +163,36 @@ def initialise_canopy_layers(
     #        The other models rely on it
 
     # Check that layers do not already exist
-    if ("leaf_area_index" in data) or ("layer_heights" in data):
-        msg = "Cannot initialise canopy layers, already present"
+    layers_to_create = [
+        "layer_heights",
+        "leaf_area_index",
+        "layer_fapar",
+        "layer_absorbed_irradiation",
+    ]
+
+    layers_found = set(layers_to_create).intersection(data.data.variables)
+    if layers_found:
+        msg = (
+            f"Cannot initialise canopy layers, already "
+            f"present: {','.join([str(x) for x in layers_found])}"
+        )
         LOGGER.critical(msg)
         raise InitialisationError(msg)
 
     # TODO - These layer roles desperately need to be set up in _one_ place!
     layer_roles = set_layer_roles(n_canopy_layers, n_soil_layers)
     layer_shape = (len(layer_roles), data.grid.n_cells)
 
-    # Set the layers
-    data["leaf_area_index"] = DataArray(
-        data=np.full(layer_shape, fill_value=np.nan),
-        dims=("layers", "cell_id"),
-        coords={
-            "layers": np.arange(len(layer_roles)),
-            "layer_roles": ("layers", layer_roles),
-            "cell_id": data.grid.cell_id,
-        },
-    )
-    data["layer_heights"] = DataArray(
-        data=np.full(layer_shape, fill_value=np.nan),
-        dims=("layers", "cell_id"),
-        coords={
-            "layers": np.arange(len(layer_roles)),
-            "layer_roles": ("layers", layer_roles),
-            "cell_id": data.grid.cell_id,
-        },
-    )
+    for each_layer in layers_to_create:
+        # Set the layers
+        data[each_layer] = DataArray(
```

Comment:
It's not an index though, it's a name 😄 Pedantry aside, (`layer_name` would work fine), I often use `each` or `this` in loops. It doesn't seem unclear to me - is there more than a stylistic reason to avoid it.

Reply:
Ah, I see! Fair enough. Nope, it was purely a stylistic suggestion.